lib/plsql/jdbc_connection.rb

```diff
@@ -348,7 +348,7 @@ def plsql_to_ruby_data_type(metadata)
         [Java::OracleSql::CLOB, nil]
       when "BLOB"
         [Java::OracleSql::BLOB, nil]
-      when "NUMBER"
+      when "NUMBER", "BINARY_FLOAT", "BINARY_DOUBLE"
```

Comment:
I tested on Ruby 1.9.3 and got:
```
irb(main):016:0> [0.3].pack("F").unpack("F")
=> *[0.30000001192092896]*
```
And on Oracle:
```
sql> DECLARE
  a BINARY_FLOAT;
BEGIN
  a := 0.3;
  DBMS_OUTPUT.PUT_LINE(TO_CHAR(A,'0.9999999999999999'));
END;
[2016-02-22 23:13:26] completed in 20ms
[2016-02-22 23:13:26]  *0.3000000120000000*
```
So it's not what we want to get.
One more interesting resource I'll try is:
https://github.com/jgoizueta/float-formats